[core][dashboard] use ray node id instead of ip for profilinglink

yancanmao opened the pull request:
<!-- Thank you for your contribution! Please review https://github.com/ray-project/ray/blob/master/CONTRIBUTING.rst before opening a pull request. -->

<!-- Please add a reviewer to the assignee section when you create a PR. If you don't have the access to it, we will shortly find a reviewer and assign them to your PR. -->

## Why are these changes needed?

Profiling links (StackTrace/CPU Profiling/Memory Profiling) currently use IP as the identifier for nodes. This becomes problematic in environments like hostnetwork, where multiple nodes can share the same IP. This can cause confusion and incorrect identification of nodes, especially when monitoring and profiling performance.
To address this, we propose using Node ID as the consistent identifier for nodes instead of IP. The Node ID is unique per node and will avoid the issues introduced by multiple nodes sharing the same IP address.
Changes:

Updated ProfilingLink to use Node ID for identifying nodes, instead of IP.
Adjusted relevant parts of the codebase to ensure consistency.

## Related issue number

Closes #55437

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a
           method in Tune, I've added it in `doc/source/tune/api/` under the
           corresponding `.rst` file.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [x] Release tests
   - [ ] This PR is not tested :(

Commit:
[core][dashboard] use ray node id instead of ip for profilinglink

Signed-off-by: Mao Yancan <yancan.mao@bytedance.com>

Commit:
reformat

Signed-off-by: Mao Yancan <yancan.mao@bytedance.com>

yancanmao committed:
Merge branch 'master' into master

jjyao reviewed (commented):
LGTM.

@alanwguo could you review the javascript files?

yancanmao committed:
Update python/ray/dashboard/modules/reporter/reporter_head.py

Co-authored-by: Jiajun Yao <jeromeyjj@gmail.com>
Signed-off-by: Mao Yancan <maoyancan@u.nus.edu>

yancanmao committed:
Update python/ray/dashboard/modules/reporter/reporter_head.py

Co-authored-by: Jiajun Yao <jeromeyjj@gmail.com>
Signed-off-by: Mao Yancan <maoyancan@u.nus.edu>

yancanmao committed:
Update python/ray/dashboard/modules/reporter/reporter_head.py

Co-authored-by: Jiajun Yao <jeromeyjj@gmail.com>
Signed-off-by: Mao Yancan <maoyancan@u.nus.edu>

yancanmao committed:
Update python/ray/dashboard/modules/reporter/reporter_head.py

Co-authored-by: Jiajun Yao <jeromeyjj@gmail.com>
Signed-off-by: Mao Yancan <maoyancan@u.nus.edu>

yancanmao commented:
> Test failure:
> 
> ```
> 
> [2025-08-09T01:54:04Z] #13 80.10 [eslint]
> --
>   | [2025-08-09T01:54:04Z] #13 80.10 src/pages/node/NodeRow.tsx
>   | [2025-08-09T01:54:04Z] #13 80.10   Line 230:5:  'ip' is assigned a value but never used  @typescript-eslint/no-unused-vars
> ```

Thanks for pointing this out. This eslint failure comes from an existing unused variable in NodeRow.tsx and is not introduced by the changes in this PR. Should I include a small cleanup commit here to remove the unused ip variable?

jjyao commented:
> Should I include a small cleanup commit here to remove the unused ip variable?

Yes, you can have a separate PR to clean it up and I'll review it.

Commit:
Edit unused ip in NodeRow

Signed-off-by: Mao Yancan <yancan.mao@bytedance.com>

yancanmao commented:
> > Should I include a small cleanup commit here to remove the unused ip variable?
> 
> Yes, you can have a separate PR to clean it up and I'll review it.



> > Should I include a small cleanup commit here to remove the unused ip variable?
> 
> Yes, you can have a separate PR to clean it up and I'll review it.

Sorry for the confusion, I double-checked the code and realized my earlier comment was inaccurate.
The `ip` variable in `NodeRow.tsx` is unused because we switched to use `nodeId`.
Therefore, we no longer need to destructure `ip` from `node`. 
I have updated this PR to remove the ip variable to fix the eslint warning.

jjyao commented:
Test failure: `[2025-08-11T09:34:20Z] //python/ray/tests:test_dashboard_profiler                               FAILED in 3 out of 3 in 145.1s`

Lint failure:

```

[2025-08-11T05:28:01Z] Checking formatting...
--
  | [2025-08-11T05:28:02Z] [warn] src/common/ProfilingLink.tsx
  | [2025-08-11T05:28:06Z] [warn] Code style issues found in the above file(s). Forgot to run Prettier?
  | [2025-08-11T05:28:21Z] 🚨 Error: The command exited with status 1

```

Commit:
fix ut for dashboard profiler

Signed-off-by: Mao Yancan <yancan.mao@bytedance.com>

Commit:
reformat test_dashboar_profiler

Signed-off-by: Mao Yancan <yancan.mao@bytedance.com>

yancanmao commented:
> Test failure: `[2025-08-11T09:34:20Z] //python/ray/tests:test_dashboard_profiler FAILED in 3 out of 3 in 145.1s`
> 
> Lint failure:
> 
> ```
> 
> [2025-08-11T05:28:01Z] Checking formatting...
> --
>   | [2025-08-11T05:28:02Z] [warn] src/common/ProfilingLink.tsx
>   | [2025-08-11T05:28:06Z] [warn] Code style issues found in the above file(s). Forgot to run Prettier?
>   | [2025-08-11T05:28:21Z] 🚨 Error: The command exited with status 1
> ```

Thanks for the comments, I have updated test_dashboard_prifiler.py and ProfilingLink.tsx to pass premerge.

yancanmao committed:
Merge branch 'ray-project:master' into master

jjyao commented:
```

[2025-08-22T07:47:20Z] TS2339: Property 'rayletId' does not exist on type 'Address'.
--
  | [2025-08-22T07:47:20Z]     656 \|                       <CpuProfilingLink
  | [2025-08-22T07:47:20Z]     657 \|                         pid={pid}
  | [2025-08-22T07:47:20Z]   > 658 \|                         nodeId={address?.rayletId}
  | [2025-08-22T07:47:20Z]         \|                                          ^^^^^^^^
  | [2025-08-22T07:47:20Z]     659 \|                         type=""
  | [2025-08-22T07:47:20Z]     660 \|                       />
  | [2025-08-22T07:47:20Z]     661 \|                       <br />

```

yancanmao committed:
Merge branch 'master' into master

Commit:
update rayletId to nodeId for consistency

Signed-off-by: Mao Yancan <yancan.mao@bytedance.com>

jjyao commented:
@yancanmao could you create a separate PR to fix the lint issue.

@alanwguo could you review the js changes?

Commit:
change pydoclin-baseline.txt to fix pydocint check error

Signed-off-by: Mao Yancan <yancan.mao@bytedance.com>

alanwguo reviewed (commented):
javascript changes look OK, but we should maintain backwards compatibility for reporter_head APIs.

jjyao reviewed (changes requested):
Keep ip for backward compatibility

Commit:
1. update pydoclint-baseline.txt to avoid lint error. 2. backword compatibility support

Signed-off-by: Mao Yancan <yancan.mao@bytedance.com>

yancanmao commented on python/ray/dashboard/modules/reporter/reporter_head.py:
```diff
@@ -413,20 +413,20 @@ async def get_traceback(self, req: aiohttp.web.Request) -> aiohttp.web.Response:
         """
         pid = req.query.get("pid")
-        ip = req.query.get("ip")
+        node_id_hex = req.query.get("node_id")
```
Thanks for the comment, I have added back IP to get stub for backwards-compatibility.

yancanmao commented:
> Keep ip for backward compatibility

I have added back the IP param for backward compatibility.  
Regarding the pydoclint failure, I noticed that we maintain a `pydoclint-baseline.txt` to exempt existing issues. Since the comment style in these functions was intentionally kept as-is (in this pr: https://github.com/ray-project/ray/pull/52974), I made a minimal update to `pydoclint-baseline.txt` so that my changes align with the current style and the checks pass.

Commit:
ut update

Signed-off-by: Mao Yancan <yancan.mao@bytedance.com>
